src/components/FullCourt/index.tsx

```diff
@@ -38,6 +42,10 @@ const FullCourt = () => {
     ref.current.y(0);
   }, [canvasStates.resetState]);
 
+  useEffect(() => {
+    updateCourtStage(dispatch, ref, rulerState);
+  }, [canvasStates.selectedColor]);
+

```

Comment:
1. In very beginning, I tried to move dispatch and rulerState to uploadImage.js. But it reported errors, because updateCourtStage() is not in a component, not in a hook.
2. I found this bug today, and it is fixed in [fix: cc-23 border change bug fixed, add bg to img] (ffea543).